### PR TITLE
timeout: fix boosted member handling, add logging

### LINKED ISSUE #216: [BUG] Timeout cog fails when member is boosting server
**Describe the bug**

Timeout cog fails when member is boosting server.

The timeout role is added, but the member's roles are not stripped because the boosted role cannot be removed by anyone, regardless of permissions.

The issue is not immediately clear to the operator either, so some enhanced error handling would be pertinent here.

**To Reproduce**
1. Run `[p]timeout <member>`, where `<member>` is a guild member who is boosting.
2. See error, see that timeout role is added but other roles are not stripped.

**Expected behavior**

User is added to timeout role, other roles are stripped after being saved to be re-added later, per the expected functionality of the cog.

**Screenshots**

**Additional context**

To fix:
* Find boosted role during init, save ID. Can be found from the [`premium_subscriber_role` attribute](https://discordpy.readthedocs.io/en/v1.7.3/api.html#discord.Guild.premium_subscriber_role) of the guild.
* Ensure the boosted role's ID is not present in the list of roles to remove before starting removal.

Alternatively, the boosted role can also be identified based on the [`is_premium_subscriber` attribute](https://discordpy.readthedocs.io/en/v1.7.3/api.html#discord.Role.is_premium_subscriber) of a role object.

### PR DESCRIPTION
**Pull request by tigattack**
# Initial Checklist
- [x] Has @tigattack / @issy been added as a reviewer?
- [x] If applicable, have the relevant project(s), milestone(s), and label(s) been applied?
- [x] If applicable, have you added details of the cog to the readme as per [README.md](https://github.com/rHomelab/LabBot-Cogs/blob/main/README.md#cog-summaries)?

# Details
**Does this resolve an issue?**
Resolves #216

### Features / Fixes
* Fixes #216 by adding logic to handle boosted members
* Add logging
* Add `[p]timeoutset list` command aliases.

### Breaking Changes
* Adds breaking change which causes \<issue\>.

## Additional
I'm bored of this template and will probably simplify it a ton at some point. Thanks for coming to my TED Talk.

### CODE REVIEW

**Comment by tigattack** [comment]
Probably just look at 230919e for a review of the actual fix here, it's a cluttered diff.

As always, I welcome the most scathing of reviews 🙏

File: timeout/timeout.py [hunk 1]
```diff
@@ -254,10 +294,20 @@ async def timeout(self, ctx: commands.Context, user: discord.Member, *, reason="
...
 
+        # Create a list containing the timeout role so we can
+        # add the boost role to it if the user is a booster.
+        # This is necessary since you cannot remove the boost
+        # role, so we must ensure we avoid attempting to do so.
+        booster_role = ctx.guild.premium_subscriber_role
+        if user in ctx.guild.premium_subscribers:
+            timeout_roleset = [timeout_role, booster_role]
+        else:
+            timeout_roleset = [timeout_role]
```
Comment (issy) [inline timeout/timeout.py:305]:
```suggestion
        timeout_roleset = {timeout_role}
        if booster_role in user.roles:
            timeout_roleset.add(booster_role)
```
May as well use a set here

File: timeout/timeout.py [hunk 2]
```diff
@@ -254,10 +294,20 @@ async def timeout(self, ctx: commands.Context, user: discord.Member, *, reason="
...
+
         # Check if user already in timeout.
-        # Remove & restore if so, else timeout.
-        if user.roles == [everyone_role, timeout_role]:
+        # Remove & restore if so, else add to timeout.
+        if sorted(user.roles) == sorted([everyone_role] + timeout_roleset):
```
Comment (issy) [inline timeout/timeout.py:309]:
```suggestion
        if set(user.roles) == {everyone_role} | {timeout_roleset}:
```
Set comparison seems more straightforward here...

File: timeout/timeout.py [hunk 3]
```diff
@@ -254,10 +294,20 @@ async def timeout(self, ctx: commands.Context, user: discord.Member, *, reason="
...
             await self.timeout_remove(ctx, user, reason)
 
         else:
-            await self.timeout_add(ctx, user, reason, timeout_role)
+            await self.timeout_add(ctx, user, reason, timeout_role, timeout_roleset)
```
Comment (issy) [inline timeout/timeout.py:313]:
```suggestion
            await self.timeout_add(ctx, user, reason, timeout_role, list(timeout_roleset))
```
Maybe pass it back as a list here. Or don't. I don't know

File: timeout/timeout.py [hunk 4]
```diff
@@ -57,27 +59,49 @@ async def report_handler(self, ctx: commands.Context, user: discord.Member, acti
...
+                f"Something went wrong while trying to add user {self.target} to timeout.\n" +
+                f"Current roles: {user.roles}\n" +
+                f"Attempted new roles: {timeout_roleset}"
+            )
+            return
+
         # Store the user's current roles
         user_roles = []
         for role in user.roles:
             user_roles.append(role.id)
```
Comment (issy) [inline timeout/timeout.py:79]:
```suggestion
        user_roles = [r.id for r in user.roles]
```
It just makes sense 🤷‍♀️

**Review by tigattack — COMMENTED** [review]
Thanks Issy, much appreciated as always 🙏

**Comment by tigattack** [comment]
Completely forgot about this, whoops. Would you be so gracious as to provide me with your most generous approval please, @issy?

**Review by issy — APPROVED** [review]
LGTM

**Pull request merged**